lld/COFF/InputFiles.cpp

```diff
@@ -1019,9 +1020,16 @@ void ImportFile::parse() {
 
   // Read names and create an __imp_ symbol.
   StringRef buf = mb.getBuffer().substr(sizeof(*hdr));
-  StringRef name = saver().save(buf.split('\0').first);
+  StringRef nameBuf = buf.split('\0').first, name;
+  if (isArm64EC(hdr->Machine)) {
+    if (std::optional<std::string> demangledName =
+            getArm64ECDemangledFunctionName(nameBuf))
+      name = saver().save(*demangledName);
+  }
+  if (name.empty())
+    name = saver().save(nameBuf);
   StringRef impName = saver().save("__imp_" + name);
-  buf = buf.substr(name.size() + 1);
+  buf = buf.substr(nameBuf.size() + 1);
```

Comment:
Hm, this seems to be a preexisting issue, but the rename of `name` to `nameBuf` and the newly added code inbetween highlights a logical stumbling block here.

Instead of doing a `buf.substr()` here at this point, what if we'd do this upfront at the top:
```
auto split = buf.split('\0');
StringRef nameBuf = split.first;
buf = split.second;
```
Then we don't need to do the `substr()` at a later point, where the index feels much less clearly related to the split.

Reply:
Good point, I changed that. I think that with the additional `split` variable, we no longer need a separate `nameBuf`.

Reply:
Nice, that does seem to have helped a fair bit with readability of the code here!